### ISSUE TITLE
Return predictions with .evaluate()

### ISSUE THREAD

**Issue by OverLordGoldDragon** [tensorflow/tensorflow#35349]
I work with several custom metrics I compute from model's inference-mode predictions; problem is, this requires me to hand-code loss computation to avoid running `.evaluate()` along `.predict()` (doubling validation time), which may include weight and activity regularizers. 

It'd be quite helpful to have an option for `model.evaluate()` to return predictions along computed metrics (e.g. `evaluate(return_predictions=True)`).

*NOTE*: this is a **feature request**, concerning the [`evaluate`](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/engine/training.py#L730) method of `keras.Model` or `tf.keras.Model`.

**Comment by gadagashwini** [comment]
@OverLordGoldDragon, Could you provide the Tensorflow version that you are using and also provide if any code snippet. Thanks!

**Comment by OverLordGoldDragon** [comment]
@gadagashwini This is a feature request, not a bug report - `model` is a `keras.Model` or a `tensorflow.keras.Model` instance, which has the `evaluate()` method defined [here](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/engine/training.py#L730)

**Comment by pavithrasv** [comment]
Thank you for the request @OverLordGoldDragon. We do not plan to add this to the built in loops. You can implement this in a custom evaluate loop by keeping track of the batch prediction values.
